adapters/handlers/grpc/v1/prepare_reply.go

```diff
@@ -311,6 +317,56 @@ func extractGroup(raw any, searchParams dto.GetParams, scheme schema.Schema, use
 		return nil, "", fmt.Errorf("cannot parse _additional %v", groupRaw)
 	}
 
+	ret := &pb.GroupByResult{
+		Name:            group.GroupedBy.Value,
+		MaxDistance:     group.MaxDistance,
+		MinDistance:     group.MinDistance,
+		NumberOfObjects: int64(group.Count),
+	}
+
+	groupedGenerativeResults := ""
+	if generativeSearchEnabled {
+		generate, ok := addAsMap["generate"]
+		if !ok {
+			return nil, "", errors.New("No results for generative search despite a search request. Is a the generative module enabled?")
+		}
+
+		if ok {
+			generateFmt, ok := generate.(*models.GenerateResult)
+			if !ok {
+				return nil, "", errors.New("could not cast generative result additional prop")
+			}
+			if generateFmt.Error != nil {
+				return nil, "", generateFmt.Error
+			}
+
+			if generateFmt.SingleResult != nil && *generateFmt.SingleResult != "" {
+				ret.Generative = &pb.GenerativeReply{Result: *generateFmt.SingleResult}
+			}
+
+			// grouped results are only added to the first object for GQL reasons
+			// however, reranking can result in a different order, so we need to check every object
+			// recording the result if it's present assuming that it is at least somewhere and will be caught
+			if generateFmt.GroupedResult != nil && *generateFmt.GroupedResult != "" {
```

Comment:
should we add a check to the level above to make sure that it was really caught?